emperor/support_files/js/view-controller.js

```diff
@@ -253,6 +239,75 @@ define([
      * application. This argument is passed by reference.
      */
     this.decompViewDict = decompViewDict;
+
+    return this;
+  }
+  EmperorViewController.prototype = Object.create(
+      EmperorViewControllerABC.prototype);
+  EmperorViewController.prototype.constructor = EmperorViewControllerABC;
+
+  /**
+   * Retrieves the metadata field currently being controlled
+   *
+   * @return {String} A key corresponding to the active decomposition view.
+   */
+  EmperorViewController.prototype.getActiveDecompViewKey = function() {
+    return this.activeViewKey;
+  };
+
+  /**
+   * Changes the metadata column name to control.
+   *
+   * @param {String} k Key corresponding to active decomposition view.
+   */
+  EmperorViewController.prototype.setActiveDecompViewKey = function(k) {
+    // FIXME: this should be validated against decompViewDict i.e. we should be
```

Comment:
I can look through in a moment

On Jul 28, 2016 08:29, "Antonio Gonzalez" <notifications@github.com> wrote:

> In emperor/support_files/js/view-controller.js
> <https://github.com/biocore/emperor/pull/528#discussion_r72644778>:
>
> > +  /**
> > +   * Retrieves the metadata field currently being controlled
> > +   *
> > +   * @return {String} A key corresponding to the active decomposition view.
> > +   */
> > +  EmperorViewController.prototype.getActiveDecompViewKey = function() {
> > +    return this.activeViewKey;
> > +  };
> > +
> > +  /**
> > +   * Changes the metadata column name to control.
> > +   *
> > +   * @param {String} k Key corresponding to active decomposition view.
> > +   */
> > +  EmperorViewController.prototype.setActiveDecompViewKey = function(k) {
> > +    // FIXME: this should be validated against decompViewDict i.e. we should be
>
> Just noticed this, should this be an issue?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/biocore/emperor/pull/528/files/6b118f7f6dd9154c17ff1dfb19d2ad7b90b32b97#r72644778>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAc8si9SNDo1TteFVUWBKcwmcbxM2SQFks5qaMrJgaJpZM4JWvVz>
> .
>